src/main.rs

```diff
@@ -129,54 +134,39 @@ pub fn start(mut os_input: Box<dyn OsApi>, opts: Opt) {
         opts.debug,
     );
 
+    if let Some(layout_path) = opts.layout {
```

Comment:
@imsnif Why was this in the event loop in the thread in the first place? Am I missing something?

Reply:
The original thought was that we either
1. Have a layout, in which case we need to create panes as the layout specified.
2. Don't have a layout in which case we need to create one pane.

I think if we want to do the validation outside, we can pass an Option<Layout> instead of using the Opt and it'll do the same thing. (or use `and_then` on opt.layout if you're feeling fancy)

Reply:
The validation is happening now in the `Layout` constructor, do you want to move it to here?

Reply:
@imsnif Fancy is my middle name

Reply:
Whatever you feel would work best

Reply:
@imsnif Since the layout is from a static file I don't think that we need to re-parse it in every iteration of the event loop.
I think that parsing it once will be better for both the logic and performance - unless the initial layout/file content can change then it *has* to be in the loop.